miden-lib/asm/miden/tx.masm

```diff
@@ -195,7 +195,55 @@ export.execute_foreign_procedure
     # reset the current account data offset to the native offset (2048)
     exec.kernel_proc_offsets::end_foreign_context_offset
     # => [offset, <values returned from the foreign procedure>]
-    
+
     syscall.exec_kernel_proc
     # => [<values returned from the foreign procedure>]
 end
+
+#! Updates the transaction expiration block number.
+#!
+#! The input block_height_delta is added to the block reference number in order to output an upper
+#! limit at which the transaction will be considered valid (not expired).
+#! This value can be later decreased, but not increased.
+#!
+#! Inputs: [block_height_delta, ...]
+#! Output: [...]
+#!
+#! Where:
+#! - block_height_delta is the desired expiration time delta (1 to 0xFFFF).
+export.update_expiration_block_num
```

Comment:
I think we want `update_expiration_block_delta` here (ie, in the user-facing procedure), unless I misunderstood @bobbinth. If this is the case, comment on line 203 should probably be updated as well.

Reply:
Yes, exactly. From the user's standpoint, we are operating with the delta rather than the block number. So, the user should be setting and getting the current delta. The fact that we are storing it as expiration block number internally, I think, is an implementation details.